lib/heroku/helpers/addons/resolve.rb

```diff
@@ -85,16 +82,23 @@ def resolve_addon(identifier, &filter)
           if identifier =~ RESOURCE
             name = $1
 
-            addon = begin
-              get_addon(name)
+            begin
+              addon = get_addon(name)
+              return [addon] if addon
             rescue Heroku::API::Errors::Forbidden
               # treat permission error as no match because there might exist a
               # resource on someone else's app that has a name which
               # corresponds to a service name that we wish to check below (e.g.
               # "memcachier")
+            rescue Heroku::API::Errors::RequestFailed => e
+              error_id = (json_decode(e.response.body.to_s) || {})["id"]
+              if error_id == "multiple_matches"
+                # ignore; name-based matches are never ambiguous, so this
+                # identifier is probably a service/plan
```

Comment:
Yeah it's tricky. The next piece will prompt for `app` which will error and ask for `--app`, so I think it always ends up in the app-scoped case when using those identifiers. We *could* have it be an ambiguous match for all the `papertrail`s across all app, but the way to handle that will be to ask for an app anyway, so it's really just moving the error. In any case, #1706 should make all of this more sane. What do you think?